### PR TITLE
Rancheros

### PR DESCRIPTION
**Pull request by jerryB1**
Pull request from the fork.

### CODE REVIEW

File: source/usage/rancheros.rst [hunk 1]
```diff
@@ -1,3 +1,57 @@
...
+RancherOS, a lightweight, secure Linux distribution, built from containers to run containers well.
+
+Preparations
+============
+Create a ``clouds.yaml`` and ``secure.yaml`` file (see :ref:`Authentication` chapter).
+
+
+Cloud-config
+============
+Cloud-config is a declarative configuration file format supported by many Linux distributions and is the primary configuration mechanism for ``RancherOS``. A Linux OS supporting cloud-config will invoke a cloud-init process during startup to parse the cloud-config file and configure the operating system. ``RancherOS`` runs its own cloud-init process in a system container. The cloud-init process will attempt to retrieve a cloud-config file from a variety of data sources. Once cloud-init obtains a cloud-config file, it configures the Linux OS according to the content of the cloud-config file.
```
Comment (berendt) [inline source/usage/rancheros.rst]: Please do not copy text from other projects.

File: source/usage/rancheros.rst [hunk 2]
```diff
@@ -1,3 +1,57 @@
...
+
+* https://rancher.com/docs/os/v1.x/en/installation/configuration/#cloud-config
+
```
Comment (berendt) [inline source/usage/rancheros.rst:22]: remove newline

File: source/usage/rancheros.rst [hunk 3]
```diff
@@ -1,3 +1,57 @@
...
+* https://cloudinit.readthedocs.io/en/latest/
+
```
Comment (berendt) [inline source/usage/rancheros.rst]: remove newline

File: source/usage/rancheros.rst [hunk 4]
```diff
@@ -1,3 +1,57 @@
...
+* https://cloudinit.readthedocs.io/en/latest/topics/modules.html
+
+
+Openstack Config drive
+======================
+If we change the settings from the ``RancherOS``, we need to enable the config drive parameter. Pass ``--config-drive`` to the ``openstack server create`` command (https://docs.openstack.org/nova/queens/user/config-drive.html).
```
Comment (berendt) [inline source/usage/rancheros.rst]: Always link the `latest` or `master` version of OpenStack docs

File: source/usage/rancheros.rst [hunk 5]
```diff
@@ -1,3 +1,57 @@
...
+
+  .. code-block:: none
```
Comment (berendt) [inline source/usage/rancheros.rst]: Type is `console` and not `none`

File: source/usage/rancheros.rst (hunk 4 above)
Comment (berendt) [inline source/usage/rancheros.rst]: Add a newline between text and title

File: source/usage/rancheros.rst [hunk 6]
```diff
@@ -1,3 +1,57 @@
...
+     
+     openstack server create --config-drive true ....
+
+
+Examples
+========
+* Create a ``rancher.txt`` file and define the ``hostname``, ``etc hosts`` and ``ssh authorized keys``. Try somthing like this:
```
Comment (berendt) [inline source/usage/rancheros.rst:40]: Add a newline between text and title

File: source/usage/rancheros.rst [hunk 7]
```diff
@@ -1,3 +1,57 @@
...
+  
+  .. code-block:: none
```
Comment (berendt) [inline source/usage/rancheros.rst]: Type is `yaml` and not `none`

File: source/usage/rancheros.rst [hunk 8]
```diff
@@ -1,3 +1,57 @@
...
+Cloud-config is a declarative configuration file format supported by many Linux distributions and is the primary configuration mechanism for ``RancherOS``. A Linux OS supporting cloud-config will invoke a cloud-init process during startup to parse the cloud-config file and configure the operating system. ``RancherOS`` runs its own cloud-init process in a system container. The cloud-init process will attempt to retrieve a cloud-config file from a variety of data sources. Once cloud-init obtains a cloud-config file, it configures the Linux OS according to the content of the cloud-config file.
+
+* https://rancher.com/docs/os/v1.x/en/installation/configuration/#cloud-config
+
+* https://cloudinit.readthedocs.io/en/latest/
+
+* https://cloudinit.readthedocs.io/en/latest/topics/modules.html
+
+
+Openstack Config drive
```
Comment (berendt) [inline source/usage/rancheros.rst]: OpenStack configuration drive

File: source/usage/rancheros.rst [hunk 9]
```diff
@@ -1,3 +1,57 @@
...
+
+     manage_etc_hosts: yes
+
+     ssh_authorized_keys:
+       - ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDoI70hvzGAiwxUqQQIfypVCSUxfx0zFZKgzBUkjIznUeDl6+rSwUtphS.....
+
+
+* Execute the ``openstack server create`` command:
+
+  .. code-block:: none
```
Comment (berendt) [inline source/usage/rancheros.rst]: `console` and not `none`

File: source/usage/rancheros.rst [hunk 10]
```diff
@@ -1,3 +1,57 @@
...
+
+     openstack server create --config-drive true --image "RancherOS 1.5.4" --flavor 2C-4GB-40GB --nic net-id=55bd2e08-428d-484b-9ac3-8ce5882e1c68 --security-group linux --max 1 --user-data /home/user/rancher.txt testinstance
```
Comment (berendt) [inline source/usage/rancheros.rst]:
```
openstack server create \
    --config drive true \
    [...]
```

**Review by berendt — CHANGES_REQUESTED** [review]

File: source/usage/rancheros.rst (hunk 4 above)
Comment (jerryB1) [reply source/usage/rancheros.rst]: no master or latest from the config-drive found.

**Pull request merged**